euphoria-core/src/test/java/cz/seznam/euphoria/core/client/operator/JoinTest.java

```diff
@@ -57,6 +57,32 @@ public void testBuild() {
     assertEquals(Join.Type.INNER, join.getType());
   }
 
+  @Test
+  public void testBuild_OutputValues() {
+    Flow flow = Flow.create("TEST");
+    Dataset<String> left = Util.createMockDataset(flow, 2);
+    Dataset<String> right = Util.createMockDataset(flow, 3);
+
+    Dataset<String> joined = Join.named("JoinVals")
+        .of(left, right)
+        .by(String::length, String::length)
+        .using((String l, String r, Collector<String> c) -> {
+          // no-op
+        })
+        .outputValues();
+
+    assertEquals(flow, joined.getFlow());
+    assertEquals(2, flow.size());
+
+    Join join = (Join) flow.operators().iterator().next();
+    assertEquals(flow, join.getFlow());
+    assertEquals("JoinVals", join.getName());
+    assertNotNull(join.leftKeyExtractor);
+    assertNotNull(join.rightKeyExtractor);
+    // FIXME: how to compare?
```

Comment:
How to compare output?
assertEquals does not work because equals is not overidden.

Reply:
`DatasetAssert#unorderedEquals()`

Reply:
i don't think there should be output comparison in this test. That would belong to operator-testkit

Reply:
simple instantce comparison should be enough here (`assertTrue(x == y)`)